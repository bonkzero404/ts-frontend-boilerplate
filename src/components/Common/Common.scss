@import '../../scss/variable.scss';

.container {
  width: 100%;
  height: 100vh;
}

.box {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  text-align: center;

  a {
    text-decoration: none;
    color: $color-carrot;
    &:hover {
      color: $color-sun;
      text-decoration: none;
    }
  }

  .headText {
    color: $color-sassy;
    font-size: 5em;
    text-align: center;
    margin-top: 30px;
    margin-bottom: 20px;
    line-height: .7em;
  }

  .logo {
    animation-name: spin;
    animation-duration: 5000ms;
    animation-iteration-count: infinite;
    animation-timing-function: linear;
  }
}

@keyframes spin {
    from {
        transform:rotate(0deg);
    }
    to {
        transform:rotate(360deg);
    }
}
