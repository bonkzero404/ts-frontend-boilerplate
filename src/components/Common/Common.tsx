import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import { State } from '../../store/reducers';
import * as Style from './Style.scss';
import ImageLogo from '../../assets/images/logo.png';

const mapStateToProps = (state: State) => ({
  pathname: state.router.location.pathname,
});

type TProps = ReturnType<typeof mapStateToProps>;

const Common: React.FunctionComponent<TProps> = (props: TProps) => {
  return (
    <div className={Style.container}>
      <div className={Style.box}>
        <LazyLoadImage className={Style.logo} src={ImageLogo} alt="Logo" />
        <h1 className={Style.headText}>React JS</h1>
        <p>
          Horay <Link to="/counter">Click Me</Link> to play counter
        </p>
        <p>You are here :{props.pathname}</p>
      </div>
    </div>
  );
};

export default hot(connect(mapStateToProps, {})(Common));
