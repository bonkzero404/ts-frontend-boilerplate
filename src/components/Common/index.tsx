import React, { Suspense, FunctionComponent, lazy } from 'react';

const LoadableComponent = lazy(() =>
  import(/* webpackPrefetch: true */ './Common')
);

const Loading = () => <div>Loading</div>;

const Common: FunctionComponent<{}> = () => (
  <Suspense fallback={<Loading />}>
    <LoadableComponent />
  </Suspense>
);

export default Common;
