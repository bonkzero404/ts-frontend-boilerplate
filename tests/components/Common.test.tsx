import { mountComponent, rendererComponent } from '../utils/test-utils';
import Common from '../../src/components/Common/Common';

describe('<Common />', () => {
  let wrapper: any;

  const props: any = {};

  it('defines the component', () => {
    wrapper = mountComponent(Common, props);
    expect(wrapper).toBeDefined();
  });

  it('defines the snapshoot component', () => {
    const tree = rendererComponent(Common, props).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
