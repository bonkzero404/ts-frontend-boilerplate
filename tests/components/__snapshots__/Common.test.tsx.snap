// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Common /> defines the snapshoot component common 1`] = `
<div>
  <div>
    <span
      className=""
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
    <h1>
      React JS
    </h1>
    <p>
      Horay 
      <a
        href="/counter"
        onClick={[Function]}
      >
        Click Me
      </a>
       to play counter
    </p>
    <p>
      You are here :
      /
    </p>
  </div>
</div>
`;
